code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -40,10 +40,8 @@
 	doomsday_device = DOOM
 	doomsday_device.start()
 	verbs -= /mob/living/silicon/ai/proc/nuke_station
-	for(var/obj/item/weapon/pinpointer/point in pinpointer_list)
-		for(var/mob/living/silicon/ai/A in ai_list)
-			if((A.stat != DEAD) && A.nuking)
-				point.the_disk = A //The pinpointer now tracks the AI core
+	for(var/obj/item/weapon/pinpointer/P in pinpointer_list)
```

Comment:
the idea is that if we know the contents of the list we don't want byond ```istype()```ing it every for iteration (it's probably not a real ```istype()``` likely it's an internal one in C++ but it still adds up)